registry/protocol/protocol.go

```diff
@@ -44,7 +44,7 @@ import (
 )
 
 var (
-	regProtocol *registryProtocol
+	regProtocol = newRegistryProtocol()
```

Comment:
is `lazy loading` better for `newRegistryProtocol` ?

Reply:
use the double-check to get or create the instance?

Reply:
I think so , dubbo-check or lock or once?

Reply:
done